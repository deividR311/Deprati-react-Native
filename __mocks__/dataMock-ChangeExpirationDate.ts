export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
