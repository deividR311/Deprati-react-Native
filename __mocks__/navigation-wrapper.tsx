import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
