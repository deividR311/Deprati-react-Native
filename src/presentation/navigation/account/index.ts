export * from './interfaces';
export * from './account.navigator';
export * from './my-returns/my-returns.navigator';
