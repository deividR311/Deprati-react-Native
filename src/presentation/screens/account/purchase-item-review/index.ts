export * from './purchase-item-review';
