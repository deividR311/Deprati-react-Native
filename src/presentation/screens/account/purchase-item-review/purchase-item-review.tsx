import * as React from 'react';
import {
  KeyboardAvoidingView,
  Platform,
  SafeAreaView,
  ScrollView,
  Text,
  View
} from 'react-native';
import { useFormik } from 'formik';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { COLORS } from '../../../../application/common/colors';
import { FontStyles } from '../../../../application/common/fonts';
import { reviewScoreValidation } from '../../../../application/common/yup-validations/auth.validations';
import { LocalStorageKey } from '../../../../application/state-manager/services/localstorage';
import { useLocalStorage } from '../../../../application/state-manager/services/localstorage/useLocalStorage';
import { MainButton } from '../../../common-components/buttons/Button';
import InputBase from '../../../common-components/inputs/InputBase';
import { Popup } from '../../../common-components/poppup';
import ErrorText from '../../../common-components/texts/ErrorText';
import ScoreIndicator, {
  ScoreRange
} from '../order-details/components/score-indicator';
import { OrderReviewProps } from '../order-details/order-details.interface';
import { Styles } from './purchase-item-review.stylesheet';
import {
  ReviewForm,
  usePurchaseItemReview
} from './usePurchaseItemReview.hook';

export const PurchaseItemReview: React.FC<OrderReviewProps> = ({ route }) => {
  const { review } = route.params;
  const [score, setScore] = React.useState(review?.score ?? 1);

  const { doSubmit, isLoading, showPopup } = usePurchaseItemReview(
    route.params
  );

  const formik = useFormik<ReviewForm>({
    ...FORMIK_SETTINGS,
    validateOnMount: false,
    validateOnChange: false,
    validateOnBlur: false,

    onSubmit: values => doSubmit(values)
  });

  const {
    localStorageData: {
      [LocalStorageKey.User]: { name: USER_FULL_NAME }
    }
  } = useLocalStorage();

  const handleVerifyEnable = () => {
    if (formik.isValid && handleVerifyValues() && !isLoading) return false;
    return true;
  };

  const handleVerifyValues = () => {
    if (
      formik.values.score.length &&
      formik.values.score !== '0' &&
      formik.values.title.length &&
      formik.values.comment.length
    )
      return true;
    return false;
  };

  const hasScore = (): boolean => {
    return !!review?.score;
  };

  const handleScoreChange = (scoreSelect: number) => {
    if (hasScore()) return;
    setScore(scoreSelect);
    formik.setFieldValue('score', `${scoreSelect}`);
  };

  const handleSendReview = async () => {
    formik.submitForm();
  };

  return (
    <SafeAreaView style={[Styles.container]}>
      <ScrollView
        style={[Styles.content]}
        contentContainerStyle={Styles.contentContainer}>
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : undefined}
          keyboardVerticalOffset={50}>
          <View style={[Styles.profile__container]}>
            <Icon
              selectionColor={COLORS.GRAYBRAND}
              name="account-circle"
              size={32}
              color={COLORS.GRAYDARK60}
            />
            <Text style={[FontStyles.Caption, Styles.profile__container_name]}>
              {USER_FULL_NAME}
            </Text>
          </View>
          <View style={[Styles.score]}>
            <ScoreIndicator
              onChangeRating={handleScoreChange}
              score={score as ScoreRange}
              size={32}
            />
          </View>

          {/* FORM */}
          <InputBase
            testID="review-title"
            dense
            label="Título de la reseña"
            keyboardType="default"
            returnKeyType="next"
            value={formik.values.title}
            error={!!formik.errors.title}
            disabled={isLoading || hasScore()}
            onBlur={formik.handleBlur}
            onChangeText={(text: string) =>
              formik.setFieldValue('title', text.trim())
            }
          />
          {!!formik.errors.title && <ErrorText text={formik.errors.title} />}
          <View style={[Styles.container_short_spacer]} />
          <InputBase
            testID="review-comment"
            keyboardType="numbers-and-punctuation"
            dense
            multiline={true}
            numberOfLines={6}
            style={{
              height: Platform.select({
                ios: 100,
                android: undefined
              }),
              justifyContent: 'flex-start'
            }}
            label="Escribe una reseña"
            value={formik.values.comment}
            error={!!formik.errors.comment}
            disabled={isLoading || hasScore()}
            onBlur={formik.handleBlur}
            onChangeText={(text: string) =>
              formik.setFieldValue('comment', text.trim())
            }
          />
          {!!formik.errors.comment && (
            <ErrorText text={formik.errors.comment} />
          )}
        </KeyboardAvoidingView>
      </ScrollView>
      <KeyboardAvoidingView
        keyboardVerticalOffset={Platform.OS === 'ios' ? 110 : undefined}
        behavior={Platform.OS === 'ios' ? 'padding' : undefined}>
        <MainButton
          title="PUBLICAR"
          style={Styles.button__send}
          onPress={handleSendReview}
          // disabled={isLoading || hasScore() || !formik.isValid}
          disabled={handleVerifyEnable() || hasScore()}
          showActivityIndicator={isLoading}
        />
      </KeyboardAvoidingView>
      <Popup
        icon="check-circle"
        iconColor={COLORS.GREENOK}
        hideButton={true}
        title={'La reseña ha sido\nañadida con éxito'}
        textContent="Podrás verla en las reseñas
         del producto"
        textContentStyle={FontStyles.Center}
        visible={showPopup}
      />
    </SafeAreaView>
  );
};

const FORMIK_SETTINGS = {
  initialValues: {
    title: '',
    comment: '',
    score: ''
  },

  initialErrors: {},
  validationSchema: reviewScoreValidation
};
