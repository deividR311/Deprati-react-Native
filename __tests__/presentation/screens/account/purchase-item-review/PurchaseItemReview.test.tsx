import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { PurchaseItemReview } from '../../../../../src/presentation/screens/account/purchase-item-review';

let screenTest: any;

describe('PurchaseItemReview', () => {
  beforeEach(() => {
    screenTest = render(<NavigationWrapper screen={PurchaseItemReview} />);
  });
  it('renders correctly in Snapshot', () => {
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
