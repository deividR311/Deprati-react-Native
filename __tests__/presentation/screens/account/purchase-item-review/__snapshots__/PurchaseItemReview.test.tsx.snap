// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseItemReview renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "flex": 1,
              },
            ]
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingHorizontal": 16,
                "paddingTop": 16,
              }
            }
            style={
              Array [
                Object {},
              ]
            }
          >
            <View>
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "paddingBottom": 0,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignContent": "center",
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    selectionColor="#E6E6E6"
                    style={
                      Array [
                        Object {
                          "color": "#979797",
                          "fontSize": 32,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 12,
                          "fontWeight": "500",
                          "letterSpacing": 0.8,
                        },
                        Object {
                          "marginLeft": 8,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginBottom": 16,
                        "marginRight": 16,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginTop": 8,
                      }
                    }
                  >
                    <View
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#FF9016",
                              "fontSize": 32,
                            },
                            Object {
                              "marginLeft": 8,
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#E6E6E6",
                              "fontSize": 32,
                            },
                            Object {
                              "marginLeft": 8,
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#E6E6E6",
                              "fontSize": 32,
                            },
                            Object {
                              "marginLeft": 8,
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#E6E6E6",
                              "fontSize": 32,
                            },
                            Object {
                              "marginLeft": 8,
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#E6E6E6",
                              "fontSize": 32,
                            },
                            Object {
                              "marginLeft": 8,
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#979797",
                          "borderRadius": 4,
                          "borderWidth": 1,
                        },
                      ]
                    }
                    testID="text-input-outline"
                  />
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "paddingBottom": 0,
                          },
                          Object {
                            "minHeight": 48,
                            "paddingTop": 8,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="none"
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                            ],
                            "zIndex": 3,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "bottom": 4,
                              "left": 10,
                              "maxHeight": 2,
                              "opacity": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 6,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                              "width": 12,
                            }
                          }
                        />
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "color": "transparent",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 18,
                              "maxWidth": -28,
                              "opacity": 0,
                              "paddingHorizontal": 0,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 29,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                                Object {
                                  "scaleY": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Título de la reseña
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          onLayout={[Function]}
                          style={
                            Object {
                              "color": "#263238",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 0,
                              "opacity": 0,
                              "paddingHorizontal": 14,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 28,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Título de la reseña
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.54)",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 0,
                              "opacity": 0,
                              "paddingHorizontal": 14,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 28,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Título de la reseña
                        </Text>
                      </View>
                      <TextInput
                        accessibilityIgnoresInvertColors={true}
                        accessibilityLabel="review-title"
                        autoComplete="off"
                        editable={true}
                        keyboardType="default"
                        maxFontSizeMultiplier={1.5}
                        multiline={false}
                        nativeID="review-title"
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        placeholderTextColor="#979797"
                        returnKeyType="next"
                        selectionColor="#979797"
                        style={
                          Array [
                            Object {
                              "flexGrow": 1,
                              "margin": 0,
                              "paddingHorizontal": 14,
                              "zIndex": 1,
                            },
                            Object {
                              "height": 40,
                            },
                            Object {
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            },
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "textAlign": "left",
                              "textAlignVertical": "center",
                            },
                            false,
                            Array [
                              Object {},
                            ],
                          ]
                        }
                        testID="review-title"
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": 8,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "justifyContent": "flex-start",
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#979797",
                          "borderRadius": 4,
                          "borderWidth": 1,
                        },
                      ]
                    }
                    testID="text-input-outline"
                  />
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "paddingBottom": 0,
                          },
                          Object {
                            "minHeight": 100,
                            "paddingTop": 8,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="none"
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                            ],
                            "zIndex": 3,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "bottom": 4,
                              "left": 10,
                              "maxHeight": 2,
                              "opacity": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 6,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                              "width": 12,
                            }
                          }
                        />
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "color": "transparent",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 18,
                              "maxWidth": -28,
                              "opacity": 0,
                              "paddingHorizontal": 0,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 59,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                                Object {
                                  "scaleY": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Escribe una reseña
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          onLayout={[Function]}
                          style={
                            Object {
                              "color": "#263238",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 0,
                              "opacity": 0,
                              "paddingHorizontal": 14,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 58,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Escribe una reseña
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.54)",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "left": 0,
                              "opacity": 0,
                              "paddingHorizontal": 14,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 58,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                        >
                          Escribe una reseña
                        </Text>
                      </View>
                      <TextInput
                        accessibilityIgnoresInvertColors={true}
                        accessibilityLabel="review-comment"
                        autoComplete="off"
                        editable={true}
                        keyboardType="numbers-and-punctuation"
                        maxFontSizeMultiplier={1.5}
                        multiline={true}
                        nativeID="review-comment"
                        numberOfLines={6}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        placeholderTextColor="#979797"
                        selectionColor="#979797"
                        style={
                          Array [
                            Object {
                              "flexGrow": 1,
                              "margin": 0,
                              "paddingHorizontal": 14,
                              "zIndex": 1,
                            },
                            Object {
                              "height": 100,
                            },
                            Object {
                              "paddingBottom": 10,
                              "paddingTop": 10,
                            },
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "textAlign": "left",
                              "textAlignVertical": "top",
                            },
                            false,
                            Array [
                              Object {},
                            ],
                          ]
                        }
                        testID="review-comment"
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            onLayout={[Function]}
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <View
              accessibilityLabel=""
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 7,
                  "marginHorizontal": 16,
                  "marginTop": 1,
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingVertical": 10,
                  "width": "94%",
                }
              }
              testID=""
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": undefined,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  PUBLICAR
                </Text>
              </View>
            </View>
          </View>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            transparent={true}
            visible={true}
          >
            <RCTSafeAreaView
              emulateUnlessSupported={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 300,
                    "justifyContent": "center",
                    "padding": 16,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "opacity": 0,
                    "paddingHorizontal": 24,
                    "paddingVertical": 16,
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  onPress={[Function]}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#00CF14",
                        "fontSize": 72,
                      },
                      Object {
                        "alignSelf": "center",
                        "paddingBottom": 24,
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <Text
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "letterSpacing": 0.15,
                        "lineHeight": 28,
                        "textAlign": "center",
                      },
                      Object {
                        "paddingBottom": 16,
                      },
                      Object {},
                    ]
                  }
                >
                  La reseña ha sido
añadida con éxito
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Podrás verla en las reseñas del producto
                </Text>
              </View>
            </RCTSafeAreaView>
          </Modal>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
